tests/drivers/hwinfo/api/src/main.c

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2019 Alexander Wachter
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <hwinfo.h>
+#include <ztest.h>
+#include <strings.h>
+
+/*
+ * @addtogroup t_hwinfo_get_device_id_api
+ * @{
+ * @defgroup t_hwinfo_get_device_id test_hwinfo_get_device_id
+ * @brief TestPurpose: verify device id get works
+ * @details
+ * - Test Steps
+ *   -# Read the ID
+ *   -# Check if to many bytes are written to the buffer
+ *   -# Check if UID is plausible
+ * - Expected Results
+ *   -# Device uid with correct length should be written to the buffer.
+ * @}
+ */
+
+#define BUFFER_LENGTH 17
+#define BUFFER_CANARY 0xFF
+
+#if defined(CONFIG_HWINFO_DEVICE_ID_STM32) || defined(CONFIG_HWINFO_DEVICE_ID_NRF)
```

Comment:
Why is this ifdef needed?

Reply:
Removed it. @MaureenHelm showed me the trick